### PR TITLE
attestations: Simplify the model.

### PR DESCRIPTION
**Pull request by MarkLodato**
- Add an Overview section explaining the relationship between raw signatures and attestations.
- Remove the policy diagram since it's not germane to this doc.
- Remove the Type fields from the diagram and instead mention them as a recommendation.
- Remove most of the requirements since they're not super helpful. Instead, make the Summary section explain the most important bits.
- Move "Materials" to the Predicate layer and rename to "Link", since that is where it logically belongs and matches English.
- Add a bunch of future items, based on feedback.

### CODE REVIEW

**Comment by TomHennen** [comment]
I'm not sure the 'type' field is just an implementation detail.  It seems like a key feature that allows for expandability in the future while avoiding some of the problems with 'raw signing'.

**Comment by MarkLodato** [comment]
I just pushed a new commit that adds a bit stronger statements about the typing. Does that help? My thinking is that it's a bit confusing to have the type as a specific field in the diagram. Instead, we can just make the "type" part of the predicate, at least in the model. WDYT?

File: attestations.md [hunk 1]
```diff
@@ -30,82 +59,38 @@ Example in English:
...
-    -   **Predicate:** Describes arbitrary properties of the subject using a
-        type-specific schema.
+-   **Artifact:** Immutable blob of data, usually identified by cryptographic
+    content hash. Examples: file content, git commit, Docker image. May also
+    include a mutable locator, such as a package name or URI.
+-   **Attestation:** Authenticated, machine-readable metadata about one or more
+    software artifacts. MUST contain at least:
+    -   **Envelope:** Authenticates the message. At a minimum, it contains:
+        -   **Message:** Content (statement) of the attestation. The message
+            type SHOULD be authenticated and unambiguous to avoid confusion
```
Comment (TomHennen) [inline attestations.md:69]: Why just SHOULD vs MUST for authenticating the type?

Comment (MarkLodato) [reply attestations.md:69]: My thinking was this: if something violates a MUST, we don't consider it an attestation at all. So here, if there is some format that meets all the requirements but this one, would we still call it an attestation? I thought yes, thus SHOULD. WDYT?

Comment (TomHennen) [reply attestations.md:69]: Sounds fine. :)

**Pull request merged**